compiler_opt/rl/log_reader.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Reader for the simple log format (non-protobuf).
+
+Refer to (in llvm repo) to llvm/lib/Analysis/models/log_reader.py
+which is used there for testing.
+"""
+
+import ctypes
+import dataclasses
+import json
+import math
+
+from typing import BinaryIO, List, Optional
+
+_element_types = {
+    'float': ctypes.c_float,
+    'double': ctypes.c_double,
+    'int8_t': ctypes.c_int8,
+    'uint8_t': ctypes.c_uint8,
+    'int16_t': ctypes.c_int16,
+    'uint16_t': ctypes.c_uint16,
+    'int32_t': ctypes.c_int32,
+    'uint32_t': ctypes.c_uint32,
+    'int64_t': ctypes.c_int64,
+    'uint64_t': ctypes.c_uint64
+}
+
+
+@dataclasses.dataclass(frozen=True)
+class TensorSpec:
+  name: str
+  port: int
+  shape: List[int]
+  element_type: type
```

Comment:
is the intended type annotation "type"?

Reply:
yes, it's a type (e.g. `ctypes.c_float`)